src/partial17.lib/partial17/Partial.test.cpp

```diff
@@ -11,16 +12,48 @@ using namespace partial17;
 
 using meta17::type;
 
-TEST(Partial, basic) {
-    auto x = Partial<char, int, float>{}; //
-    auto y = x; // copy constructor
-    x = y; // copy operator
-    x = std::move(y); // move operator
-    auto z = Partial<char, int, float>{std::move(x)}; // move construct
+TEST(Partial, construction) {
+    using P = Partial<char, int, float>;
+    auto p0 = P{};
+    ASSERT_FALSE(p0.has<0>());
+    ASSERT_FALSE(p0.has<1>());
+    ASSERT_FALSE(p0.has<2>());
+    // TODO CK: Should this be possible? - crashes, would suggest a graceful value for production

```

Comment:
This is by intention. `get<N>` is like a `cast`. You have to know that the type is there, otherwise ghosts will hunt you at night. If we throw exceptions here, the resulting program would suffer a huge performance hit.

Reply:
Is there no way to prevent an memory access crash when trying to read uninitialized memory? Since there is the `m_bitset` to check, maybe we can return a reference to some static default for production mode while crashing with an assertion in debug mode?

Reply:
Checking every time kills the optimizer.
The solution would be to rename `get` into `cast` or `as` and add a `get` that is checked. Like `operator[]` and `at` on a `std::vector`. That way we have options at the usage side.